projects/js-packages/shared-extension-utils/src/block-editor-actions.js

```diff
@@ -0,0 +1,55 @@
+import { dispatch } from '@wordpress/data';
+import domReady from '@wordpress/dom-ready';
+
+const JETPACK_EDITOR_ACTION = 'jetpack-editor-action';
+
+/**
+ * Get the Jetpack Editor action from the URL.
+ *
+ * @return {string | null} The Jetpack Editor action.
+ */
+export function getJetpackEditorAction() {
+	const url = new URL( window.location.href );
+
+	return url.searchParams.get( JETPACK_EDITOR_ACTION );
+}
+
+/**
+ * Remove the Jetpack Editor action from the URL.
+ *
+ */
+export function removeJetpackEditorAction() {
+	const url = new URL( window.location.href );
+	url.searchParams.delete( JETPACK_EDITOR_ACTION );
+	window.history.replaceState( null, '', url.toString() );
+}
+
+/**
+ * Handle the Jetpack Editor action.
+ *
+ * @param {{sidebarToOpen?: string; onAction?: (action:string) => void; removeQueryArg?: boolean}} args - Arguments.
+ */
+export function handleJetpackEditorAction( {
+	sidebarToOpen = 'jetpack-sidebar/jetpack',
```

Comment:
However, I don't think this should be concerned with sidebars.

Reply:
I have moved that sidebar logic back to publicize-components.